### PR TITLE
Set default geo mirror for Arch Linux

### PR DESCRIPTION
**Pull request by Foxboron**
Signed-off-by: Morten Linderud <morten.linderud@mullvad.net>

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
With this, can we drop the logic to select a random mirror from the mirrorlist in install_arch()? That doesn't really work all that well since we often get a less than perfect mirror.

**Comment by Foxboron** [comment]
Oh, I missed there was logic for that. I'll rm that.

**Comment by Foxboron** [comment]
@DaanDeMeyer Should be good. The only issue would be archlinux32 I *think*, but the code would not have worked there anyway.

File: mkosi/__init__.py [hunk 1]
```diff
@@ -2710,19 +2710,8 @@ def install_arch(args: MkosiArgs, root: Path, do_run_build_script: bool) -> None
...
-        with urllib.request.urlopen(
-            "https://www.archlinux.org/mirrorlist/?country=all&protocol=https&ip_version=4&use_mirror_status=on"
-        ) as r:
-            mirrors = r.readlines()
-            uncommented = [line.decode("utf-8")[1:] for line in mirrors]
-            with mirrorlist.open("w") as f:
-                f.writelines(uncommented)
-            server = f"Include = {mirrorlist}"
+        # This should not happen, but for good measure.
+        raise ValueError("No repository mirror has been selected.")
```
Comment (DaanDeMeyer) [inline mkosi/__init__.py:2714]: We use the `die()` function for this.

**Review by DaanDeMeyer — CHANGES_REQUESTED** [review]

**Review by DaanDeMeyer — APPROVED** [review]

**Comment by behrmann** [comment]
Merging, since all the Arch tests have passed and this doesn't touch anything else.

**Pull request merged**